src/inmanta/server/services/orchestrationservice.py

```diff
@@ -760,7 +760,10 @@ async def put_version(
         async with data.Resource.get_connection() as con:
             async with con.transaction():
                 # Acquire a lock that conflicts with the lock acquired by put_partial.
-                await data.Resource.lock_table(data.TableLockMode.ROW_EXCLUSIVE, connection=con)
+                await con.execute(
```

Comment:
Take advisory lock instead of table lock:
- no deadlocks possible due to table locking order
- don't lock full table, only for this environment (sort of, the full uuid is too long to use as key, so we only use part of the timestamp).